alfresco-search/src/main/java/org/alfresco/solr/component/RewriteFacetParametersComponent.java

```diff
@@ -337,12 +341,76 @@ private void copyOtherQueryParams(ModifiableSolrParams fixed, SolrParams params)
     }
 
     /**
-     * @param fixed ModifiableSolrParams
-     * @param params SolrParams
-     * @param paramName String
-     * @param fieldMappings HashMap<String, String>
+     * Ensure the mincount for all given facets is at least 1 to prevent exposing sensitive buckets to users without permission.
+     *
+     * @param fixed The updated params object.
+     * @param params The original params object.
+     * @param paramName The name of the mincount parameter to rewrite (e.g. "facet.mincount" or "facet.pivot.mincount").
+     * @param fieldMappings A list of mappings from Alfresco property names to Solr field names.
+     * @param facetNames The list of all fields for this facet type.
+     *
+     * Note: Set to protected to allow unit testing.
+     */
+    protected void rewriteMincountFacetFieldOption(ModifiableSolrParams fixed, SolrParams params, String paramName,
+                Map<String, String> fieldMappings, List<String> facetNames)
+    {
+        rewriteFacetFieldOptions(fixed, params, paramName, fieldMappings);
+
+        // Ensure that the min count is at least 1.
+        List<String> found = new ArrayList<>();
+
+        // Update any existing mincount entries for facets.
+        Map<String, Integer> updatedValues = new HashMap<>();
+        for (Iterator<String> it = fixed.getParameterNamesIterator(); it.hasNext(); /**/)
+        {
+            String name = it.next();
+            if (name.startsWith("f."))
+            {
+                if (name.endsWith("." + paramName))
+                {
+                    String facetName = name.substring(2, name.length() - paramName.length() - 1);
+                    if(fieldMappings.containsKey(facetName))
+                    {
+                        facetName = fieldMappings.get(facetName);
+                    }
+
+                    // If mincount is set then it must be an integer.
+                    int value = Integer.valueOf(fixed.get(name));
+                    // Don't directly edit the params while we're iterating through the entries.
+                    updatedValues.put("f." + facetName + "." + paramName, Math.max(value, 1));
+                    found.add(facetName);
+                }
+            }
+        }
+        // Now we've finished iterating, update the fixed parameters.
+        for (Map.Entry<String, Integer> entry : updatedValues.entrySet())
+        {
+            fixed.set(entry.getKey(), entry.getValue());
+        }
+
+        // Add a mincount entry for any facets that didn't already have one.
+        if (facetNames != null)

```

Comment:
At this point using the facet.mincount parameter is recommended.
No need in iterating on facet that didn't have one

Reply:
I wouldn't normally expect the iteration over an empty collection to be expensive though.